File: Data Structures & Algorithms/jump-game/submission-11.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_reach = 0;
        int i = 0;
        while (i < (int)nums.size() and max_reach < (int)nums.size() - 1) {
            if (max_reach < i) {
                return false;
            } else {
                max_reach = max(max_reach, nums[i] + i);
                i++;
            }
        }
        return i < (int)nums.size();
    }
};

File: Data Structures & Algorithms/house-robber/submission-11.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        prev2, prev1 = 0, nums[0]
        for i in range(1, len(nums)):
            curr = max(prev1, prev2 + nums[i])
            prev2 = prev1
            prev1 = curr
        return max(prev2, prev1)